bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/zk/ZKMetadataDriverBase.java

```diff
@@ -260,6 +268,46 @@ public synchronized LedgerManagerFactory getLedgerManagerFactory()
         return lmFactory;
     }
 
+    public CompletableFuture<Void> disableHealthCheck() {
+        CompletableFuture<Void> createResult = new CompletableFuture<>();
+        try {
+            zk.create(disableHealthCheckPath, BookKeeperConstants.EMPTY_BYTE_ARRAY, acls, CreateMode.PERSISTENT);
+            createResult.complete(null);
+        } catch (KeeperException.NodeExistsException nodeExistsException) {
+            log.debug("health check already disable!");
+            createResult.complete(null);
+        } catch (Exception e) {
+            createResult.completeExceptionally(e);
+        }
+        return createResult;
+    }
+
+    public CompletableFuture<Void>  enableHealthCheck() {
+        CompletableFuture<Void> deleteResult = new CompletableFuture<>();
+
+        try {
+            zk.delete(disableHealthCheckPath, -1);
+            deleteResult.complete(null);
+        } catch (KeeperException.NoNodeException noNodeException) {
+            log.debug("health check already enabled!");
+            deleteResult.complete(null);
+        } catch (Exception e) {
+            deleteResult.completeExceptionally(e);
+        }
+        return deleteResult;
+    }
+
+    public CompletableFuture<Boolean> isHealthCheckEnabled() {
+        CompletableFuture<Boolean> enableResult = new CompletableFuture<>();
+        try {
+            boolean isEnable = (null == zk.exists(disableHealthCheckPath, false));
```

Comment:
please use the async API

Reply:
I will fix

Reply:
like this?

    public CompletableFuture<Void> disableHealthCheck() {
        CompletableFuture<Void> createResult = new CompletableFuture<>();
        zk.create(disableHealthCheckPath, BookKeeperConstants.EMPTY_BYTE_ARRAY, acls,
                CreateMode.PERSISTENT, new AsyncCallback.StringCallback() {
            
            public void processResult(int rc, String path, Object ctx, String name) {
                if (KeeperException.Code.OK.intValue() == rc) {
                    createResult.complete(null);
                } else if (KeeperException.Code.NODEEXISTS.intValue() == rc) {
                    log.debug("health check already disable!");
                    createResult.complete(null);
                } else {
                    createResult.completeExceptionally(KeeperException.create(KeeperException.Code.get(rc), path));
                }
            }
        }, null);

        return createResult;
    }

Reply:
public CompletableFuture<Boolean> isHealthCheckEnabled() {
        CompletableFuture<Boolean> enableResult = new CompletableFuture<>();
        zk.exists(disableHealthCheckPath, false, new AsyncCallback.StatCallback() {
            @Override
            public void processResult(int rc, String path, Object ctx, Stat stat) {
                if (KeeperException.Code.OK.intValue() == rc) {
                    enableResult.complete(false);
                } else {
                    enableResult.complete(true);
                }
            }
        }, null);

        return enableResult;
    }